packages/@aws-cdk/aws-redshift-alpha/lib/private/database-query-provider/table.ts

```diff
@@ -1,34 +1,40 @@
 /* eslint-disable-next-line import/no-unresolved */
 import * as AWSLambda from 'aws-lambda';
+import { Column } from '../../table';
 import { executeStatement } from './redshift-data';
 import { ClusterProps, TableAndClusterProps, TableSortStyle } from './types';
-import { areColumnsEqual, getDistKeyColumn, getSortKeyColumns } from './util';
-import { Column } from '../../table';
+import { areColumnsEqual, getDistKeyColumn, getSortKeyColumns, makePhysicalId } from './util';
 
 export async function handler(props: TableAndClusterProps, event: AWSLambda.CloudFormationCustomResourceEvent) {
   const tableNamePrefix = props.tableName.prefix;
-  const tableNameSuffix = props.tableName.generateSuffix === 'true' ? `${event.RequestId.substring(0, 8)}` : '';
+  const tableNameSuffix = props.tableName.generateSuffix === 'true' ? `${event.StackId.substring(event.StackId.length - 12)}` : '';
```

Comment:
Yes, as this would make it possible for `DELETE` requests. The `Data` attribute is not possible to be transferred over to `DELETE` requests unfortunately, so we need to be able to persist the table name through all these events, without using the `PhysicalResourceId`. So I thought to use the `stackId` as this would persist over the lifecycle of the stack